### DISCUSSION TITLE
Email Templates: with {{ body }} BB adds some unwanted HTML to result email

### DISCUSSION THREAD

**Discussion by sergey-dme** [Budibase/budibase#1984] [Help] [answered]
Hi All,

I made my Email template which is complex and auto-adjustable depending on a viewer of email receipt (email app/device).
Then, for test purposes I inserted the code of my template in the builder - field "**Base**" and saved it. 
I  did not use` {{ body }}` in **Base** for the purpose. 
Using Automations I sent the email (just **Base** only). 
The test passed okay  - the email I received was the same as the code I inserted in **Base**.

Then I split template to "**Base**" part and "**HTML Content**" part.  added` {{ body }}` in **Base**. And some surprises came...

So, if  `{{ body }} ` is in the **Base** , budibase starts adding the some unwanted code which does not exist in **Base** and **HTML Content**, that breaks whole template and makes it unworkable.

The unwanted code BB adds in result email when `{{ body }}` is used - in **Base** this following code is absent:

`<tr>`
`<td class="email-body" width="570" cellpadding="0" cellspacing="0">`
`    <table class="email-body_inner" align="center" width="570" cellpadding="0" cellspacing="0" role="presentation">`
`      <!-- Body content -->`
`      <tr>`
`        <td class="content-cell">`
`          <div class="f-fallback">`

_(HTML Contents code is here)_

 `</div>`
`        </td>`
`      </tr>`
`    </table>`
`  </td>`
`</tr>`

**Email templates should work like this, for example:** 

1. If the **Base** code is: `<tag> {{ body }} </tag> `
2. And the **HTML Contents**  - {{ body }} part is:  `<tag2> Abc.. </tag2> `
3. **THE RESULT** email should be: `<tag><tag2> Abc.. </tag2></tag> ` without any additional code parts

Please resolve it, otherwise this is significant limitation for using user created HTML email templates.

P.S. I used Beyond Compare app to find the differences in code between two files.  Actually I spent two days investigating it.

**Comment by joebudi** [comment]
@sergey-dme thank you for looking into this. I will chat with the team re this.

**Comment by mike12345567** [comment, marked as answer]
This answer was given in the issue as well.

Hi @sergey-dme - there may be some confusion here over how email templating works, this is understandable because its not really made clear anywhere how the SMTP email action works.

What is happening here is that all email templates are two parts, a base and then based on the main purpose of the email the body gets filled in. The base is never actually sent directly. In the case of a SMTP email automation it uses the combination of the base and the custom email templates to create what is going to be sent. So if you want to just send what is in your base, you simply replace the content within the custom email template with {{ contents }}.

The idea behind this is that you build the look of your email once and then put it into the base, then simply fill the correct content for each email type into the others, this way you can get emails that all follow the same theme without much effort.

I'm thinking I might add some more explanation in this as to how it works as for SMTP automations this isn't very clear at all.

**Reply by sergey-dme** [reply]
@mike12345567 Thank your for the prompt reply. More clear now and your solution works, I'm happy now :)
I think this was my fault - just found that "unwanted html" in Email Custom section - the code which is added with {{ body }}

**Reply by mike12345567** [reply]
@sergey-dme its all good, actually brought to my attention its a little confusing, looking at adding some descriptions around each of the email templates as I think that would avoid this confusion in the future!
